### ISSUE TITLE
isinstance with type[Protocol] doesn't work now.

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#6463]
**Describe the bug**
After #434, `isinstance` is now rejecting if the second argument is `type[P]` where `P` is a protocol. Although of course `isinstance(x, P)` should be rejected, it's less clear (to me) that `type[P]` should, since it can (and did in our code) mean any class that conforms to the protocol. See the code below: `type[P]` meaningfully restricts the set of types you can pass as an argument, but still allows classes that can legally be used at runtime as the second argument to `isinstance`.  

**Code or Screenshots**

```python
from typing import Any, Protocol


class P(Protocol):
    def doesnt_matter(self) -> None:
        ...


class C(P):
    def doesnt_matter(self) -> None:
        return None


def foo1(x: Any, t: type[Any]) -> bool: # okay, even though you could pass `P`, see runtime error below
    return isinstance(x, t)


def foo2(x: Any, t: type[C]) -> bool: # okay
    return isinstance(x, t)


# this error is questionable, since the foo3 is usable with non-protocol types below
def foo3(x: Any, t: type[P]) -> bool: # error: Second argument to "isinstance" must be a class or tuple of classes
                                      # Protocol class must be @runtime_checkable to be used with instance and class checks (reportGeneralTypeIssues)

    return isinstance(x, t)


assert foo3(C(), C)
# this error is great and shows why `type[P]` is preferred to `type[Any]`. 
assert foo3(1, str) # error: Argument of type "type[str]" cannot be assigned to parameter "t" of type "type[P]" in function "foo3"
assert not foo3(1, C)
assert foo1(C(), P) # runtime TypeError: Instance and class checks can only be used with @runtime_checkable protocols
```

This is an edge case and I can understand why it might not be worth fixing. 

**VS Code extension or command-line**
Command line, 1.1.336.

**Comment by erictraut** [comment]
Yeah, I agree that it should be OK to pass a value of type `type[Protocol]` to `isinstance` even if `Protocol` isn't `@runtime_checkable`. As you point out, this opens up a small hole if you pass the actual protocol class, but I think that's a reasonable tradeoff. It's also consistent with mypy.

This change will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
